### PR TITLE
Don't display folders in restore output

### LINKED ISSUE #3281: [Bug]: Contacts folders shown in restore details results
### What happened?

The command `corso restore exchange --backup deadbeef...` (backup only contains contacts) prints out both the restored contact (santiago), as well as the restored contact folder (Contacts).  This is a regression, only the contacts should be displayed in the restore details.

```
Restoring to folder Corso_Restore_02-May-2023_17:37:29
  ID            Contact Name
  74840f9459b1  Sanitago TestContact - 2023-04-28T20:58:44.2865310Z
  ID            Display Name
  4e1276a1b6d9  Contacts
```

### Corso Version?

Corso v0.6.1

### Where are you running Corso?

any

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by ashmrtn**
#### Does this PR need a docs update or release note?

- [x] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [ ] :no_entry: No

#### Type of change

- [ ] :sunflower: Feature
- [x] :bug: Bugfix
- [ ] :world_map: Documentation
- [ ] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [ ] :broom: Tech Debt/Cleanup

#### Issue(s)

* closes #3281

#### Test Plan

- [x] :muscle: Manual
- [ ] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

**Comment by ashmrtn** [comment]
Manual output for a contacts restore (one contact restored)
```
corso restore exchange --retain-progress --backup aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee
Connecting to repository:      …
Restoring to folder Corso_Restore_15-May-2023_17:29:23
Connecting to M365:         done 
Restoring ∙ aaaaaaaaa-bbbb-cccc-…
Discovered 1 items in backup aa…
Enumerating items in repository:done
Restoring data:      2s done 
contacts 1 - Corso_Restore_15-M…
  ID            Contact Name  
  b89aa3550718  Santiago Quail
```

File: CHANGELOG.md [hunk 1]
```diff
@@ -23,6 +23,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Refined oneDrive rate limiter controls to reduce throttling errors.
 - Fix handling of duplicate folders at the same hierarchy level in Exchange. Duplicate folders will be merged during restore operations.
 - Fix backup for mailboxes that has used up all their storage quota
+- Don't show restored folders in CLI restore output. Only show restored items instead.
```
Comment (ryanfkeepers) [inline CHANGELOG.md:26]:
```suggestion
- Restored folders will no longer appear in the Restore results. Only items will get displayed.
```

**Review by ryanfkeepers — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (ashmrtn) [reply CHANGELOG.md]: thanks! guess I should put more thought into my changelog messages 😅

Comment (ryanfkeepers) [reply CHANGELOG.md]: heh, naw.  entirely a nitpick; it would've been fine.

**Pull request merged**